### PR TITLE
607 document steps to create arelease

### PR DESCRIPTION
**Pull request by daren-thomas**
@JIMENOFONSECA this pull request addresses #607 - how to create a release of the CEA. Including uploading to PyPI, testing the documentation, additional tools needed and compiling the installer for the planner's version.

If I forgot something, please let me know.

### CODE REVIEW

**Comment by daren-thomas** [comment]
How to test: 

- follow documentation here: https://github.com/architecture-building-systems/CEAforArcGIS/blob/607-document-steps-to-create-arelease/docs/how-to-create-a-new-release.rst
- does it explain everything you need to know?
- do the commands listed work?

File: CREDITS.md [hunk 1]
```diff
@@ -1,6 +1,17 @@
 Credits
```
Comment (JIMENOFONSECA) [inline CREDITS.md]: We need to include Arno in the credits of all releases

File: docs/how-to-create-a-new-release.rst [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+Where to find the current version number
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The current version number can be found in the module :py:mod:`cea` (actually, since :py:mod:`cea` is a package, you
+need to look into the file ``__init__.py``) in the variable ``__version__``.
+All code requiring knowledge of the current version number should read the version from here. In python this can be
+achieved by:
+
+.. source: python
```
Comment (JIMENOFONSECA) [inline docs/how-to-create-a-new-release.rst]: this means opening the terminal window or the anaconda prompt? maybe we want to describe it in here

File: docs/how-to-create-a-new-release.rst [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+    import cea
+    version_number = cea.__version__
+
+
+The NSIS installer uses the helper tool ``setup/get_version.exe`` to extract the version and write it to the file
```
Comment (JIMENOFONSECA) [inline docs/how-to-create-a-new-release.rst]: What does the NSIS installer stand for?

File: docs/how-to-create-a-new-release.rst [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+.. _GitHub issues milestones list: https://github.com/architecture-building-systems/CEAforArcGIS/milestones
+
+
+Update the CREDITS.md file
+--------------------------
+
+For each minor release (2.2, 2.3, ...) the ``CREDITS.md`` file needs to be updated to include all the authors that
+worked on that release.
+
+@JIMENO: Do I understand this correctly: The list is ever expanding? Then we also might need a HISTORY.md file or
```
Comment (JIMENOFONSECA) [inline docs/how-to-create-a-new-release.rst]: yep, although I am not sure if we want to do it in every minor release or in ever major. Let's say the authors are added or changed in a yearly basis? for me taht would work as well.

File: docs/how-to-create-a-new-release.rst (hunk 2 above)
Comment (daren-thomas) [inline docs/how-to-create-a-new-release.rst]: um... neither. this is python code. do you have a recommendation for stating this in a more obvious manner? I tried "in python this is achieved by"... but obviously it is not good enough?

File: docs/how-to-create-a-new-release.rst (hunk 3 above)
Comment (daren-thomas) [inline docs/how-to-create-a-new-release.rst]: addressed in next commit

File: docs/how-to-create-a-new-release.rst (hunk 4 above)
Comment (daren-thomas) [inline docs/how-to-create-a-new-release.rst]: I think the unit of credits should be a release as this is also the vehicle a feature is introduced with. But I'm open for discussions.

**Comment by daren-thomas** [comment]
@JIMENOFONSECA, I just updated the branch - jenkins is running, but maybe you would like to review again afterwards?

File: docs/how-to-create-a-new-release.rst (hunk 4 above)
Comment (JIMENOFONSECA) [inline docs/how-to-create-a-new-release.rst]: good point. let's do it that way. (the unit of credits should be a release

File: docs/how-to-create-a-new-release.rst (hunk 2 above)
Comment (JIMENOFONSECA) [inline docs/how-to-create-a-new-release.rst]: so in python means I open a console, type python and then proceed.?

**Comment by JIMENOFONSECA** [comment]
@daren-thomas i made some extra comments

**Comment by daren-thomas** [comment]
@JIMENOFONSECA wanna merge?

File: CREDITS.md (hunk 1 above)
Comment (JIMENOFONSECA) [inline CREDITS.md]: @daren-thomas could we..?

**Comment by JIMENOFONSECA** [comment]
@daren-thomas there are some comments that have not been addressed yet. could you please check? after that we are good to merge.

File: docs/how-to-create-a-new-release.rst (hunk 2 above)
Comment (daren-thomas) [inline docs/how-to-create-a-new-release.rst]: no. in python means, when you are coding a module. this  paragraph explains how to find out the current CEA version in code. if you need it. I thought it was obvious. Should I change the language to make it more obvious?

File: docs/how-to-create-a-new-release.rst (hunk 4 above)
Comment (daren-thomas) [inline docs/how-to-create-a-new-release.rst]: @JIMENOFONSECA do you mean minor release? Because I take it to mean minor release.

**Comment by daren-thomas** [comment]
OK. I think we are good to merge. @JIMENOFONSECA can you do the honors? (once the jenkins has completed?)

File: CREDITS.md (hunk 1 above)
Comment (JIMENOFONSECA) [inline CREDITS.md]: Yes we could!

File: docs/how-to-create-a-new-release.rst (hunk 2 above)
Comment (JIMENOFONSECA) [inline docs/how-to-create-a-new-release.rst]: You mean coding a module from pycharm? If so can this be specified? I do not still know where to run these comands..

File: docs/how-to-create-a-new-release.rst (hunk 4 above)
Comment (JIMENOFONSECA) [inline docs/how-to-create-a-new-release.rst]: Yes for the minor release

File: docs/how-to-create-a-new-release.rst (hunk 2 above)
Comment (daren-thomas) [inline docs/how-to-create-a-new-release.rst]: No, @JIMENOFONSECA, specifying PyCharm is not a good idea as that narrows the scope. I think the language is already super clear. If you want to find the version of CEA with python code (it doesn't matter from where!) then you use these commands. Please reread the paragraph carefully and then tell me if you still don't understand.

**Pull request merged**